packages/fx-tunnel-relayer/src/Relayer.ts

```diff
@@ -0,0 +1,126 @@
+import { Contract, EventData } from "web3-eth-contract";
+import { runTransaction, POLYGON_MESSAGE_SENT_EVENT_SIG } from "@uma/common";
+import type Web3 from "web3";
+
+export class Relayer {
+  constructor(
+    readonly logger: any,
+    readonly account: string,
+    readonly gasEstimator: any,
+    readonly maticPosClient: any,
+    readonly oracleChildTunnel: Contract,
+    readonly oracleRootTunnel: Contract,
+    readonly web3: Web3,
+    readonly polygonEarliestBlockToQuery: number
+  ) {}
+
+  // In order to receive a message on Ethereum from Polygon, `receiveMessage` must be called on the Root Tunnel contract
+  // with a proof derived from the Polygon transaction hash that was checkpointed to Mainnet.
+  async relayMessage(): Promise<void> {
```

Comment:
Consider renaming this to include that it also checks for new massages. Would it make sense to split this into two methods, one that checks for messages and a second relay method?

Reply:
I think if we split them into two methods they should be internal methods if anything but the "fetch" logic is super simple its just a single `getPastEvents` call. For now I"m just going to rename the method `fetchAndRelayMessages` which will call `getPastEvents` and then call the internal method `_relayMessage` for each message found.

Reply:
does that make sense to you?

Reply:
yup. that sounds like a reasonable middle ground to me